src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,92 @@
+import classNames from 'classnames';
+import { useCallback, useState, useMemo } from 'react';
+import debounce from 'lodash.debounce';
+
+import { Person } from '../types/Person';
+
+type Props = {
+  people: Person[],
+  onPersonSelected: (person: Person) => void,
+  delay: number;
+};
+
+function getFilteredfilteredPeople(people: Person[], query: string) {
+  return people.filter(({ name }) => {
+    const preparedName = name.toLowerCase();
+    const preparedQuery = query.trim().toLowerCase();
+
+    return preparedName.includes(preparedQuery);
+  });
+}
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  onPersonSelected,
+  delay,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay), [],
+  );
+
+  const filteredPeople = useMemo(
+    () => getFilteredfilteredPeople(people, appliedQuery),
+    [appliedQuery, query, people],
```

Comment:
Don't put unused values to the dependency list

```suggestion
    [appliedQuery, people],
```